src/ucp/core/ucp_mm.c

```diff
@@ -534,17 +534,20 @@ ucp_memh_register_internal(ucp_context_h context, ucp_mem_h memh,
     }
 
     reg_md_map = ~memh->md_map & md_map;
-    if (reg_md_map == 0) {
-        return UCS_OK;
-    }
-
     if ((context->config.ext.reg_nb_mem_types & UCS_BIT(mem_type)) &&
```

Comment:
So `mem_type` found in `reg_nb_mem_types` forces non-blocking registration, and only MD that support them.

Without that, it is still possible to manually pass `UCT_MD_MEM_FLAG_NONBLOCK`, which tries non-block on MDs that support it, else tries with blocking reg, is that correct?

Reply:
Yes, that's correct. According to this logic `reg_nb_mem_types` forces nonblock registration only and `UCT_MD_MEM_FLAG_NONBLOCK` considered us hint only.

Reply:
AFAIU the issue is that we pass `UCT_MD_MEM_FLAG_NONBLOCK` to UCT registration for cuda memory. How is that possible given this check for `context->config.ext.reg_nb_mem_types`?